src/acceptance/api/api_test.go

```diff
@@ -0,0 +1,252 @@
+		resp, err := DoAPIRequest(req)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		defer resp.Body.Close()
+
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(resp.StatusCode).Should(BeNumerically("==", 200))
+	})
+
+	Context("when no policy defined", func() {
+
+		var (
+			body      io.Reader
+			policyURL string
+		)
+
+		BeforeEach(func() {
+			policy := GenerateDynamicScaleOutPolicy(cfg, 1, 2, "memoryused", 30)
+			body = bytes.NewBuffer([]byte(policy))
+			policyURL = fmt.Sprintf("%s%s", cfg.ASApiEndpoint, strings.Replace(PolicyPath, "{appId}", appGUID, -1))
+		})
+
+		It("should succeed to create a policy", func() {
+			req, err := http.NewRequest("PUT", policyURL, body)
+			req.Header.Add("Authorization", oauthToken)
+			req.Header.Add("Content-Type", "application/json")
+
+			resp, err := DoAPIRequest(req)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			defer resp.Body.Close()
+
+			raw, err := ioutil.ReadAll(resp.Body)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(resp.StatusCode == 200 || resp.StatusCode == 201).Should(BeTrue())
+
+			var responsedPolicy *ScalingPolicy
+			err = json.Unmarshal(raw, &responsedPolicy)
+			Expect(err).ShouldNot(HaveOccurred())
+
+		})
+	})
+
+	Context("When policy is defined", func() {
+
+		var (
+			policy    string
+			policyURL string
+			body      io.Reader
+		)
+		BeforeEach(func() {
+			policy = GenerateDynamicScaleOutPolicy(cfg, 1, 2, "memoryused", 30)
+			body = bytes.NewBuffer([]byte(policy))
+
+			policyURL = fmt.Sprintf("%s%s", cfg.ASApiEndpoint, strings.Replace(PolicyPath, "{appId}", appGUID, -1))
+			req, err := http.NewRequest("PUT", policyURL, body)
+			req.Header.Add("Authorization", oauthToken)
+			req.Header.Add("Content-Type", "application/json")
+
+			resp, err := DoAPIRequest(req)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			defer resp.Body.Close()
+
+			Expect(resp.StatusCode == 200 || resp.StatusCode == 201).Should(BeTrue())
+		})
+
+		It("should succeed to delete a policy", func() {
+			req, err := http.NewRequest("DELETE", policyURL, nil)
+			req.Header.Add("Authorization", oauthToken)
+			req.Header.Add("Content-Type", "application/json")
+
+			resp, err := DoAPIRequest(req)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			defer resp.Body.Close()
+
+			Expect(resp.StatusCode == 200).Should(BeTrue())
+		})
+
+		It("should succeed to get a policy", func() {
+
+			req, err := http.NewRequest("GET", policyURL, nil)
+			req.Header.Add("Authorization", oauthToken)
+			req.Header.Add("Content-Type", "application/json")
+
+			resp, err := DoAPIRequest(req)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			defer resp.Body.Close()
+
+			raw, err := ioutil.ReadAll(resp.Body)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(resp.StatusCode == 200).Should(BeTrue())
+
+			var responsedPolicy *ScalingPolicy
+			err = json.Unmarshal(raw, &responsedPolicy)
+			Expect(err).ShouldNot(HaveOccurred())
+
+		})
+
+		It("should succeed to update a policy", func() {
```

Comment:
add test case to update policy with invalid policy.  The update will fail while the old policy is still there